src/pudl/transform/ferc1.py

```diff
@@ -694,6 +694,96 @@ def combine_axis_columns_xbrl(
+
+    column_to_check: str | None = None
+    """Name of data column to check.
+
+    This will typically be ``dollar_amount`` or ``ending_balance`` in hte
+    """
+    calculation_tolerance: float = 0.05
+    """The tolerance ratio of the off calcuations and the possible calcuations."""
+
+
+def check_table_calcuations(
+    df: pd.DataFrame,
+    tbl_meta: pd.DataFrame,
+    xbrl_factoid_name: str,
+    table_name: str,
+    params: CheckTableCalculations,
+) -> pd.DataFrame:
+    """Calculate the intra-table calculations and ensure the table is within tolerance.
+
+    Args:
+        df: processed table.
+        tbl_meta: processed table xbrl metadata.
+        xbrl_factoid_name: column name of the XBRL factoid in the processed table.
+        table_name: name of the PUDL table.
+        params: :class:`CheckTableCalculations` parameters.
+    """
+    # skip the calculations that have any components from other tables.
+    inter_table_calculated_values = list(
+        tbl_meta[
+            tbl_meta.calc_contains_components_from_other_tables
+        ].xbrl_factoid.unique()
+    )
+    if inter_table_calculated_values:
+        logger.info(
+            "Skipping calcuated values because they are inter-table calucations: "
+            f"{inter_table_calculated_values}"
+        )
+    intra_tbl_calcs = tbl_meta[
+        ~tbl_meta.calc_contains_components_from_other_tables
+        & (tbl_meta.row_type_xbrl == "calculated_value")
+    ]
+    pks = pudl.metadata.classes.Resource.from_id(table_name).schema.primary_key
+    pks_wo_factoid = [col for col in pks if col != xbrl_factoid_name]
+    calculated_dfs = []
+    for calcuated_factoid, calculation in zip(
+        intra_tbl_calcs.xbrl_factoid, intra_tbl_calcs.calculations
+    ):
+        calc_df = (
+            pd.merge(
+                df,
+                pd.DataFrame(literal_eval(calculation)),
+                left_on=xbrl_factoid_name,
+                right_on="name",
+            )
+            .assign(calculated_amount=lambda x: x[params.column_to_check] * x.weight)
+            .groupby(pks_wo_factoid, as_index=False)["calculated_amount"]
+            .sum(min_count=1)
+            .assign(**{xbrl_factoid_name: calcuated_factoid})
+        )
+        calculated_dfs.append(calc_df)
+
+    calculated_df = pd.merge(
+        df, pd.concat(calculated_dfs), on=pks, how="left", validate="m:1"
+    ).assign(
+        abs_diff=lambda x: abs(x[params.column_to_check] - x.calculated_amount),
+        rel_diff=lambda x: abs(x.abs_diff / x[params.column_to_check]),
+    )
+
+    off_df = calculated_df[
+        ~np.isclose(
+            calculated_df.calculated_amount, calculated_df[params.column_to_check]
+        )
+        & (calculated_df["abs_diff"].notnull())
+    ]
+    calced_values = calculated_df[(calculated_df.abs_diff.notnull())]
+    off_ratio = len(off_df) / len(calced_values)
+    logger.info(
```

Comment:
I would suggest changing this to a warning, otherwise it's almost impossible to find in the logging statements.